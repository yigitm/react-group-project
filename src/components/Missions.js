import React from 'react';

const Rockets = () => {
  return <h1>Under Construction Missions</h1>;
};
export default Rockets;
