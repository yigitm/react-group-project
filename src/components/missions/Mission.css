button {
  border-radius: 3px;
  font-size: 14px;
  cursor: pointer;
}

span {
  white-space: nowrap;
}

.button-status {
  padding: 0.2rem 0.3rem;
  font-weight: 600;
  border: none;
  background-color: grey;
  color: white;
  cursor: inherit;
}

.button-join {
  border: 2px rgb(0, 165, 165) solid;
  padding: 0.4rem 0.4rem;
  color: rgb(0, 165, 165);
}

.button-status-active {
  background-color: rgb(0, 165, 165);
}

.button-join-active {
  border: 2px solid red;
  color: red;
}
