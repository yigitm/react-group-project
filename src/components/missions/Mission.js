import React from 'react';
import PropTypes from 'prop-types';

const Mission = ({ mission }) => {
  const { name, description } = mission;

  return (
    <tr>
      <td className="fw-bold fs-5">{name}</td>
      <td>{description}</td>
      <td>Status</td>
      <td>Button</td>
    </tr>
  );
};
Mission.defaultProps = {
  mission: {
    id: '',
    name: '',
    description: '',
  },
};
Mission.propTypes = {
  mission: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
  }),
};
export default Mission;
