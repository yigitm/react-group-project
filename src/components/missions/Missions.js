import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Mission from './Mission';
import { getMissions } from '../../redux/missions/missions';

const Missions = () => {
  const missions = useSelector((state) => state.missionsReducer);
  const dispatch = useDispatch();

  useEffect(() => {
    const loadingMissions = async () => {
      await dispatch(getMissions());
    };
    loadingMissions();
  }, []);

  return (
    <div>
      <table className="table table-striped">
        <thead>
          <tr>
            <th className="fs-3 border" scope="col">Mission</th>
            <th className="fs-3 border" scope="col">Description</th>
            <th className="fs-3 border" scope="col">Status</th>
            <th className="fs-3 border" scope="col"> </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <Mission key={mission.mission_id} mission={mission} />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
