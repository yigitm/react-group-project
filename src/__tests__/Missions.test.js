import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../redux/configureStore';
import Mission from '../components/missions/Mission';
import Missions from '../components/missions/Missions';

describe('Testing User Interation: ', () => {
  const mission = {
    mission_id: '9D1B7E0',
    name: 'Thaicom',
    description: 'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
    reserved: true,
  };

  it('Tests Mission render', () => {
    const tree = render(<Provider store={store}><Mission key={mission.mission_id} mission={mission} /></Provider>);
    expect(tree).toMatchSnapshot();
  });

  
  it('Tests Missions render', () => {
    const tree = render(<Provider store={store}><Missions /></Provider>);
    expect(tree).toMatchSnapshot();
  });

  
  it('Tests Join/Leave Button', () => {
    render(<Provider store={store}><Mission key={mission.mission_id} mission={mission} /></Provider>);
    fireEvent.click(screen.getByText('Cancel Mission'));
    expect(screen.getByTestId("joinButton")).toHaveValue('Join Mission');
  });
});