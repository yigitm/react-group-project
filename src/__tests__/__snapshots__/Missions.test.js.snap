// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing User Interation:  Tests Mission render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr>
        <td
          class="fw-bold fs-5"
        >
          Thaicom
        </td>
        <td>
          Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
        </td>
        <td
          class="align-middle"
        >
          <button
            class="button-status button-status-active"
            type="button"
          >
            <span>
              Active Member
            </span>
          </button>
        </td>
        <td
          class="align-middle"
        >
          <button
            class="button-join button-join-active"
            data-testid="joinButton"
            type="button"
            value="Cancel Mission"
          >
            <span>
              Cancel Mission
            </span>
          </button>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr>
      <td
        class="fw-bold fs-5"
      >
        Thaicom
      </td>
      <td>
        Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
      </td>
      <td
        class="align-middle"
      >
        <button
          class="button-status button-status-active"
          type="button"
        >
          <span>
            Active Member
          </span>
        </button>
      </td>
      <td
        class="align-middle"
      >
        <button
          class="button-join button-join-active"
          data-testid="joinButton"
          type="button"
          value="Cancel Mission"
        >
          <span>
            Cancel Mission
          </span>
        </button>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing User Interation:  Tests Missions render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table
          class="table table-striped"
        >
          <thead>
            <tr>
              <th
                class="fs-3 border"
                scope="col"
              >
                Mission
              </th>
              <th
                class="fs-3 border"
                scope="col"
              >
                Description
              </th>
              <th
                class="fs-3 border"
                scope="col"
              >
                Status
              </th>
              <th
                class="fs-3 border"
                scope="col"
              >
                 
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <table
        class="table table-striped"
      >
        <thead>
          <tr>
            <th
              class="fs-3 border"
              scope="col"
            >
              Mission
            </th>
            <th
              class="fs-3 border"
              scope="col"
            >
              Description
            </th>
            <th
              class="fs-3 border"
              scope="col"
            >
              Status
            </th>
            <th
              class="fs-3 border"
              scope="col"
            >
               
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
